documentation/auto-discovery/topics/advanced-discovery-topics.rst

```diff
@@ -0,0 +1,101 @@
+.. include:: /auto-discovery/media.rst
+.. _Advanced-discovery-topics-doc:
+
+*************************
+Advanced discovery topics
+*************************
+
+====================
+Obtaining SNMP walks
+====================
+
+Hyperview uses SNMP walks to enhance device definitions, to model and support devices that are discoverable with the SNMP protocol. The **snmpwalk file** is used to simulate the device and to test definitions.
+
+It is recommended to install the applicable net-snmp package alongside the (Linux) Data Collector software. If that is not possible it can be installed on any machine that has a network line of sight to the target device.
+
+Linux
+-----
+
+The `net-snmp <http://www.net-snmp.org/>`_ utilities are available on all supported Linux distributions.
+
+On Debian-based distributions:
+
+.. code::
+
+    sudo apt install snmp
+
+On RedHat-based distributions:
+
+.. code::
+
+    sudo dnf install net-snmp-utils
+
+Windows
+-------
+
+There are multiple options.
+
+1. Use net-snmp using a UNIX environment simulator such as `Cygwin <https://www.cygwin.com/>`_ or `WSL <https://docs.microsoft.com/en-us/windows/wsl/>`_.
+
+2. Use a Linux Virtual Machine
+
+macOS
+-----
+
+You can install net-snmp on macOS using `Homebrew <https://brew.sh/>`_
+
+.. code::
+
+    brew install net-snmp
+
+Once the application is installed, the **snmpwalk** command can be used to obtain a full walk of a device. The amount of time it takes to perform the walk can vary from device to device and depends on external factors such as network speed and target device CPU load. Generally, expect the process to take a few minutes unless the device has a large amount of information such as a large network switch, in which case, it may take longer.
+
+**Example:**
+
+.. code::
+
+    snmpwalk -v2c -c public -ObentU 192.168.10.10 . > /my_home_dir/snmpwalks/newrackpdu.snmpwalk
+    #
+    # Compress the snmpwalk file
+    #
+    gzip /my_home_dir/snmpwalks/newrackpdu.snmpwalk
+
+.. note:: The **-ObentU** command option is required and important to be able to parse the output with our tools.
+
+Once the walk is obtained, then it can be transferred to Hyperview support by uploading the compressed file to the applicable support ticket.
```

Comment:
Our docs in the past for the snmpwalker tools highlighted verifying the completeness of a walk by checking the end of the file for "No more variables left in this MIB View (It is past the end of the MIB tree)" or similar before sending to Hyperview. This would save everyone step if it turns out the walk ended prematurely.

Similar to the last comment. This can also just be a link to a support forum article.